api/src/test/java/org/openmrs/calculation/MostRecentObsCalculation.java

```diff
@@ -74,4 +77,18 @@ public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Obj
 	public Concept getWhichConcept() {
 		return whichConcept;
 	}
-}
+
+	private Map<Integer, List<Obs>> getObservations(Cohort cohort, Concept concept) {
+		if (cohort == null || concept == null || cohort.getMemberships().isEmpty()) {
+			return Collections.emptyMap();
+		}
+
+		return cohort.getMemberships().stream()
```

Comment:
I find this to be a very inefficient replacement when compared to the original implementation: https://github.com/openmrs/openmrs-core/blob/1.12.x/api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientSetDAO.java#L1092-L1141

Reply:
@dkayiwa, does performance matter much in this case since it's only used in tests?

Reply:
That is a very good point and you are indeed correct. 😊